pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -124,14 +125,23 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	original, err := c.taskRunLister.TaskRuns(namespace).Get(name)
 	if errors.IsNotFound(err) {
 		// The resource no longer exists, in which case we stop processing.
-		c.Logger.Errorf("task run %q in work queue no longer exists", key)
+		c.Logger.Infof("task run %q in work queue no longer exists", key)
 		return nil
 	} else if err != nil {
+		c.Logger.Errorf("Error retreiving TaskRun %q: %s", name, err)
 		return err
 	}
 
 	// Don't modify the informer's copy.
 	tr := original.DeepCopy()
+	// If this is the first attempt to reconcile, mark the Run as in progress
+	if tr.Status.GetCondition(duckv1alpha1.ConditionSucceeded) == nil {
```

Comment:
I would probably make this like `InitizlieStatusCondition()` function and move the check for nil and setting condition inside that.

Reply:
nice, will do!!

Reply:
oh wait, i think im kind of confused - gonna move this to the other thread in this PR on initialize